src/io_uring.c

```diff
@@ -0,0 +1,92 @@
+#include "io_uring.h"
+
+#ifdef HAVE_LIBURING
+#include <liburing.h>
+
+#define IO_URING_DEPTH 256 /* io_uring instance queue depth. */
+
+static size_t io_uring_queue_len = 0;
+
+void initIOUring(void) {
+    if (server.io_uring_enabled) {
+        struct io_uring_params params;
+        struct io_uring *ring = zmalloc(sizeof(struct io_uring));
+        memset(&params, 0, sizeof(params));
+        /* On success, io_uring_queue_init_params(3) returns 0 and ring will
+         * point to the shared memory containing the io_uring queues.
+         * On failure -errno is returned. */
+        int ret = io_uring_queue_init_params(IO_URING_DEPTH, ring, &params);
+        if (ret != 0) {
+            /* Warning if user enable the io_uring in config but system doesn't support yet. */
+            serverLog(LL_WARNING, "Failed to initialize io_uring: %s", strerror(-ret));
+            zfree(ring);
+            server.io_uring = NULL;
```

Comment:
make io_uring a "local" global. I personally prefer to name globals differently than  stack variables. 
```suggestion
            _io_uring = NULL;
```

Reply:
Actually another better option is to return an opaque pointer to io_uring to the caller and let the caller deal with the storage. My main point is that we should write io_uring as if it is a library that knows nothing about the Valkey server, or as little as possible.

This would allow the io_uring client (other server code) to decide the best way to use io_uring (maybe multiple per-thread io_uring instances?)